src/test/java/io/airlift/compress/AbstractTestCompression.java

```diff
@@ -112,6 +114,33 @@ public void testDecompressWithOutputPadding(DataSet dataSet)
         assertByteArraysEqual(uncompressed, padding, uncompressedSize, uncompressedOriginal, 0, uncompressedOriginal.length);
     }
 
+    @Test(dataProvider = "data")
+    public void testDecompressionBufferOverrun(DataSet dataSet)
+    {
+        byte[] uncompressedOriginal = dataSet.getUncompressed();
+        byte[] compressed = prepareCompressedData(uncompressedOriginal);
+
+        // add padding with random bytes that we can verify later
+        byte[] padding = new byte[100];
+        ThreadLocalRandom.current().nextBytes(padding);
```

Comment:
Do you wan to run with this test like 10 times to get more randoms?

Reply:
I don't think it's worth it. Each additional run adds 2 minutes to the build. There's only a 0.4% chance of missing a 1-byte overrun in an individual test case, a 0.002% of missing a 2-byte overrun and so on.